### DISCUSSION TITLE
Does ruff read [tool.black] and [tool.isort]?

### DISCUSSION THREAD

**Discussion by mil-ad** [astral-sh/ruff#12168] [Q&A]
Does ruff read and respect `[tool.black]`, `[tool.isort]`, etc sections from `pyproject.toml` or do I need to add them to `[tool.ruff.lint.isort]` etc?

I'm collaborating on a project and don't want to force everyone to switch to ruff.

**Comment by charliermarsh** [comment]
No, we generally don't read configuration that's intended for other tools. So you'd either need to add them to your `pyproject.toml` in addition to those existing sections, or add a separate `ruff.toml` if you want.
